os/board/common/crashdump.c

```diff
@@ -238,6 +312,48 @@ static int send_fsdump(void)
 	return 0;
 }
 
+static int send_extfsdump(void)
+{
+	int i = 0;
+	int fd;
+	int index;
+	uint8_t *ptr;
+	size_t size;
+	char ext_userfs_partition[EXTUSERFS_PARTITION_NAME_SIZE + 1] = { '\0' };
+
+	/*Send ACK for EXTERNAL FSDUMP Handshake */
+	up_lowputc('A');
+
+	i = get_ext_userfs_part(&size, &index);
+	if (i < 0) {
+		return -1;
+	}
+
+	/* Send external userfs partition size to host */
+	ptr = (uint8_t *)&size;
+	for (i = 0; i < sizeof(size); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* If extracted size is 0, it menas ther eis no partition available to dump */
+	if (size == 0) {
+		return 0;
+	}
+
+	snprintf(ext_userfs_partition, EXTUSERFS_PARTITION_NAME_SIZE, "/proc/fs/smartfs/smart1p%d/dump_ext", index);
+
+	fd = open(ext_userfs_partition, O_RDONLY);
+	if (fd < 0) {
+		return -EIO;
```

Comment:
Is there any using this errno after calling send_extfsdump()? If not, I'm not sure what is the difference between returning ERROR and -errno.

Reply:
Since the tty connection is established by the HOST machine, returned ERROR signals cannot be tracked.
I still kept different error signals for consistency and ease of understanding.

Reply:
returning errno is correct, I think

Reply:
Done